source/acquaman/IDEAS/IDEAS2DScanActionController.cpp

```diff
@@ -0,0 +1,199 @@
+	connect(this, SIGNAL(cancelled()), &elapsedTime_, SLOT(stop()));
+	connect(this, SIGNAL(paused()), &elapsedTime_, SLOT(stop()));
+	connect(this, SIGNAL(resumed()), &elapsedTime_, SLOT(start()));
+	connect(this, SIGNAL(failed()), &elapsedTime_, SLOT(stop()));
+	connect(this, SIGNAL(finished()), &elapsedTime_, SLOT(stop()));
+	connect(&elapsedTime_, SIGNAL(timeout()), this, SLOT(onScanTimerUpdate()));
+}
+
+IDEAS2DScanActionController::~IDEAS2DScanActionController()
+{
+
+}
+
+void IDEAS2DScanActionController::buildScanControllerImplementation()
+{
+	AMXRFDetector *detector = 0;
+
+	if (configuration_->fluorescenceDetector().testFlag(IDEAS::Ketek))
+		detector = qobject_cast<AMXRFDetector *>(AMBeamline::bl()->exposedDetectorByName("KETEK"));
+
+	else if (configuration_->fluorescenceDetector().testFlag(IDEAS::Ge13Element))
+		detector = qobject_cast<AMXRFDetector *>(AMBeamline::bl()->exposedDetectorByName("13-el Ge"));
+
+	if (detector){
+
+		detector->removeAllRegionsOfInterest();
+
+		QList<AMDataSource *> i0Sources = QList<AMDataSource *>() << scan_->dataSourceAt(scan_->indexOfDataSource("I_0"));
+
+		AMDataSource *spectraSource = scan_->dataSourceAt(scan_->indexOfDataSource(detector->name()));
+
+		foreach (AMRegionOfInterest *region, configuration_->regionsOfInterest()){
+
+			AMRegionOfInterestAB *regionAB = (AMRegionOfInterestAB *)region->valueSource();
+			AMRegionOfInterestAB *newRegion = new AMRegionOfInterestAB(regionAB->name().remove(' '));
+			newRegion->setBinningRange(regionAB->binningRange());
+			newRegion->setInputDataSources(QList<AMDataSource *>() << spectraSource);
+			scan_->addAnalyzedDataSource(newRegion, false, true);
+			detector->addRegionOfInterest(region);
+
+			AM2DNormalizationAB *normalizedRegion = new AM2DNormalizationAB(QString("norm_%1").arg(newRegion->name()));
+			normalizedRegion->setInputDataSources(QList<AMDataSource *>() << newRegion << i0Sources);
+			normalizedRegion->setDataName(newRegion->name());
+			normalizedRegion->setNormalizationName(i0Sources.first()->name());
+			scan_->addAnalyzedDataSource(normalizedRegion, true, false);
+		}
+	}
+}
+
+void IDEAS2DScanActionController::createAxisOrderMap()
+{
+	axisOrderMap_.insert(scan_->rawData()->scanAxisAt(0).name, 1);
+	axisOrderMap_.insert(scan_->rawData()->scanAxisAt(1).name, 0);
+}
+
+AMAction3 * IDEAS2DScanActionController::createInitializationActions()
+{
+	AMListAction3 *initializationActions = new AMListAction3(new AMListActionInfo3("IDEAS XAS Initialization Stage 1", "IDEAS XAS Initialization Stage 1"), AMListAction3::Parallel);
+
+	double startEnergy = configuration_->energy();
+	double mono2d = IDEASBeamline::ideas()->mono2d()->value();
+	double braggAngle = asin(12398.4193 / mono2d / startEnergy);
+	double backlashDegrees = 4;
+
+	double deltaEnergy = (backlashDegrees / 180 * M_PI) * (mono2d * startEnergy * startEnergy * cos(braggAngle * M_PI / 180)) / (-12398.4193);
+	double backlashEnergy = startEnergy + deltaEnergy;
+
```

Comment:
In fact IDEASMonochromatorControl::move() has backlash correction built in.   As I recall there was some strange timing issue that led to intermittent scan failures, and adding this startup move solved it.